daft/iceberg/iceberg_write.py

```diff
@@ -222,13 +222,13 @@ def partitioned_table_to_iceberg_iter(
     partition_values = partitioned.partition_values()
 
     if partition_values:
-        partition_strings = partition_values_to_string(partition_values, partition_null_fallback="null").to_pylist()
+        partition_strings = partitioned.partition_values_str().to_pylist()  # type: ignore
         partition_values_list = partition_values.to_pylist()
 
         for table, part_vals, part_strs in zip(partitioned.partitions(), partition_values_list, partition_strings):
             iceberg_part_vals = {k: to_partition_representation(v) for k, v in part_vals.items()}
             part_record = IcebergRecord(**iceberg_part_vals)
-            part_path = partition_strings_to_path(root_path, part_strs)
+            part_path = partition_strings_to_path(root_path, part_strs, partition_null_fallback="null")
```

Comment:
Is the `__HIVE_DEFAULT_PARTITION__` thing not the appropriate behavior here? That's surprising to me given that I'm guessing Iceberg and Delta would probably use the same Parquet writers for Spark under the hood

Reply:
I'm matching pyiceberg behavior here but I can also check what Spark does

Reply:
Checked Spark and they also do this